src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -77,14 +63,12 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
     ProjectedDataModelInstance(
       externalId = dmi.properties
         .flatMap(_.get("externalId"))
-        .flatMap(_.asString)
+        .map(_.asInstanceOf[StringProperty].value)
```

Comment:
I think you can also use `collect { case StringProperty(externalId) => externalId }` here instead